SalesforceCommerceCloud/pwa-kit
# [Spike]Url Customization

# Description
As an outcome of `multi site support` spike #167, we want to make sure the PWA kit can provide the flexibility to url customization as opposed to fixed routing currently. 
# Solution
We will need to have a config file that contains the url key. It will contain the types of url paramters for the app. This config will be the source of truth to resolve the url params across the app.
The types would be one of three values: 
- `path`: this means  the param to be in the path
- `query_param`: this means the param will be in the query parameter
 For example:
```
{
    "app": {
        "url": {
          "locale": "query_param"
        },
    }
}
```

# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes


# How to Test-Drive This PR
- npm start
- Navigate to `localhost:300`, and notice the locale is in the path even though we don't clarify it in the routes array
- Go to `pwa-kit-config.json`, change the `locale` value from `path` to `query_param`, and reload the page. The locale should not be in the path any more when you navigate through the pages

This PR explores the idea of using `basename`, a prop in react router dom `BrowserRouter` and `StaticRouter` to the SDK to provide a prefixed basePath for all of the routes. This means we don't need to have a logic to handle the routes in the pwa kit everything the url config is changed.


# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
